tests/api2/test_core_download.py

```diff
@@ -0,0 +1,29 @@
+from os import stat
+
+from urllib.request import urlretrieve
+from middlewared.test.integration.utils.client import truenas_server
+from middlewared.test.integration.utils import call
+
+
+def test_get_download_for_config_dot_save():
+    # ping core for sanity
+    assert call('core.ping') == 'pong'
+
+
+    # set up core download
+    job_id, url = call('core.download', 'config.save', [], 'freenas.db')
+
+
+    # is the job running?
+    results = call('core.get_jobs', [['id', '=', job_id]], {"get": True})
+    assert results['state'] == 'RUNNING'
+
+
+    # download from URL
+    rv = urlretrieve(f'http://{truenas_server.ip}{url}')
```

Comment:
Why would you stat the size before the job returns `SUCCESS`?

Reply:
This (and the ping) were there and structured that way from the old test, I had assumed there was some specific reason for it. For this I was assuming that the download job is set to "running" while there's a pending URL to download from. Will update accordingly.